### PR TITLE
add tests for core module

### PR DESCRIPTION
**Pull request by sheb**

### CODE REVIEW

File: tests/unit/test_core.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+@pytest.fixture
+def known_versions(mocker):
+    versions = ["1.1", "1.2", "1.3"]
+    mocker.patch("west.core.files.get_known_versions", return_value=versions)
+
+    return versions
+
+
+def test_get_applied_versions(mocker, known_versions):
+    mocker.patch("west.core.db.get_applied_versions", return_value=["1.0", "1.1"])
```
Comment (k4nar) [inline tests/unit/test_core.py]: Do we plan to have functional tests for functions in the db module ?

Comment (sheb) [reply tests/unit/test_core.py]: I think we should test it, but no easy to do in a pure functional way

**Comment by sheb** [comment]
rebased on master

**Comment by sheb** [comment]
rebased on master

**Pull request merged**